### ISSUE TITLE
Docker hangs sporadically or on container stop

### ISSUE THREAD

**Issue by vpaul** [docker/docker#22233]
**Output of `docker version`:**

```
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 5
Images: 387
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-202:80-524292-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: 
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 14.47 GB
 Data Space Total: 107.4 GB
 Data Space Available: 16.96 GB
 Metadata Space Used: 24.58 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.123 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /docker/data/devicemapper/devicemapper/data
 Metadata loop file: /docker/data/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-85-generic
Operating System: Ubuntu 14.04.4 LTS
CPUs: 1
Total Memory: 3.661 GiB
Name: instance-1
ID: LM6M:4EWX:G5US:2SXB:6H5P:VFAL:F6C2:GDKE:CF5M:CZNV:EI5E:AEQS
Debug mode (server): true
 File Descriptors: 19
 Goroutines: 30
 System Time: 2016-04-21T13:49:16.737086008Z
 EventsListeners: 0
 Init SHA1: 4fc7c03a06c675f115b39958c13486d53df10a14
 Init Path: /usr/lib/docker/dockerinit
 Docker Root Dir: /docker/data
WARNING: No swap limit support

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

AWS 

**Steps to reproduce the issue:**
docker stop ${contianer_id} or it just hangs after a while

**Describe the results you received:**

docker ps stops working, containers are not responsive.

It is reproducible also on another instance with aufs storage driver. Debug output at hung time:

```
[37mDEBU[0m[78479] Calling GET /v1.21/containers/json           
[34mINFO[0m[78479] GET /v1.21/containers/json                   
[37mDEBU[0m[78503] Calling GET /v1.21/containers/json           
[34mINFO[0m[78503] GET /v1.21/containers/json                   
[37mDEBU[0m[78510] Calling POST /v1.21/containers/ec98ac0f50d7/stop 
[34mINFO[0m[78510] POST /v1.21/containers/ec98ac0f50d7/stop?t=10 
[37mDEBU[0m[78510] Sending 15 to ec98ac0f50d77204f711b34e5b6bc1587088eb2096f110308c098a00b7659f2d 
[34mINFO[0m[78520] Container ec98ac0f50d77204f711b34e5b6bc1587088eb2096f110308c098a00b7659f2d failed to exit within 10 seconds of SIGTERM - using the force 
[33mWARN[0m[78520] Logger didn't exit in time: logs may be truncated 
[37mDEBU[0m[78628] Calling GET /v1.21/containers/json           
[34mINFO[0m[78628] GET /v1.21/containers/json                   
[37mDEBU[0m[78996] Calling GET /v1.21/containers/json           
[34mINFO[0m[78996] GET /v1.21/containers/json                   
[34mINFO[0m[79048] === BEGIN goroutine stack dump ===
goroutine 11 [running]:
github.com/docker/docker/pkg/signal.DumpStacks()
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/signal/trap.go:67 +0xa5
github.com/docker/docker/daemon.func·030()
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/debugtrap_unix.go:18 +0x6d
created by github.com/docker/docker/daemon.setupDumpStackTrap
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/debugtrap_unix.go:20 +0x18e

goroutine 1 [chan receive, 1315 minutes]:
main.(*DaemonCli).CmdDaemon(0xc208503240, 0xc20800a020, 0x4, 0x4, 0x0, 0x0)
    /usr/src/docker/docker/daemon.go:308 +0x20a7
reflect.callMethod(0xc2084e3fb0, 0xc208c8dce8)
    /usr/local/go/src/reflect/value.go:605 +0x179
reflect.methodValueCall(0xc20800a020, 0x4, 0x4, 0x1, 0xc2084e3fb0, 0x0, 0x0, 0xc2084e3fb0, 0x0, 0x4e207f, ...)
    /usr/local/go/src/reflect/asm_amd64.s:29 +0x36
github.com/docker/docker/cli.(*Cli).Run(0xc2084e3ef0, 0xc20800a010, 0x5, 0x5, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/cli/cli.go:89 +0x38e
main.main()
    /usr/src/docker/docker/docker.go:65 +0x418

goroutine 5 [syscall]:
os/signal.loop()
    /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
    /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, 1317 minutes, locked to thread]:
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 13 [IO wait]:
net.(*pollDesc).Wait(0xc208080df0, 0x72, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208080df0, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208080d90, 0x0, 0x7f8f366c2500, 0xc20a1eb058)
    /usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*UnixListener).AcceptUnix(0xc2085141c0, 0xc20a546140, 0x0, 0x0)
    /usr/local/go/src/net/unixsock_posix.go:282 +0x56
net.(*UnixListener).Accept(0xc2085141c0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/unixsock_posix.go:293 +0x4c
github.com/docker/docker/pkg/listenbuffer.(*defaultListener).Accept(0xc2085141e0, 0x0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/listenbuffer/buffer.go:71 +0x67
net/http.(*Server).Serve(0xc20800a780, 0x7f8f31f8b608, 0xc2085141e0, 0x0, 0x0)
    /usr/local/go/src/net/http/server.go:1728 +0x92
github.com/docker/docker/api/server.(*HTTPServer).Serve(0xc208514400, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:113 +0x4d
github.com/docker/docker/api/server.func·006(0xc208514400)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:86 +0x163
created by github.com/docker/docker/api/server.(*Server).ServeAPI
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:90 +0x144

goroutine 10 [chan receive, 1317 minutes]:
github.com/docker/docker/api/server.(*Server).ServeAPI(0xc2080378c0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:94 +0x1b6
main.func·007()
    /usr/src/docker/docker/daemon.go:255 +0x3b
created by main.(*DaemonCli).CmdDaemon
    /usr/src/docker/docker/daemon.go:261 +0x1571

goroutine 16 [select, 1188 minutes]:
github.com/docker/libnetwork.(*controller).watchLoop(0xc20808c840)
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/store.go:398 +0x13f
created by github.com/docker/libnetwork.(*controller).startWatch
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/store.go:412 +0xf0

goroutine 19 [IO wait, 1317 minutes]:
net.(*pollDesc).Wait(0xc208811800, 0x72, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208811800, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).readMsg(0xc2088117a0, 0xc208cae2d0, 0x10, 0x10, 0xc208ca1220, 0x1000, 0x1000, 0xffffffffffffffff, 0x0, 0x0, ...)
    /usr/local/go/src/net/fd_unix.go:296 +0x54e
net.(*UnixConn).ReadMsgUnix(0xc208039c60, 0xc208cae2d0, 0x10, 0x10, 0xc208ca1220, 0x1000, 0x1000, 0x51, 0xc208cae094, 0x4, ...)
    /usr/local/go/src/net/unixsock_posix.go:147 +0x167
github.com/godbus/dbus.(*oobReader).Read(0xc208ca1200, 0xc208cae2d0, 0x10, 0x10, 0xc208ca1200, 0x0, 0x0)
    /usr/src/docker/vendor/src/github.com/godbus/dbus/transport_unix.go:21 +0xc5
io.ReadAtLeast(0x7f8f31fa5f30, 0xc208ca1200, 0xc208cae2d0, 0x10, 0x10, 0x10, 0x0, 0x0, 0x0)
    /usr/local/go/src/io/io.go:298 +0xf1
io.ReadFull(0x7f8f31fa5f30, 0xc208ca1200, 0xc208cae2d0, 0x10, 0x10, 0x51, 0x0, 0x0)
    /usr/local/go/src/io/io.go:316 +0x6d
github.com/godbus/dbus.(*unixTransport).ReadMessage(0xc208c80660, 0xc208bd9f20, 0x0, 0x0)
    /usr/src/docker/vendor/src/github.com/godbus/dbus/transport_unix.go:85 +0x1bf
github.com/godbus/dbus.(*Conn).inWorker(0xc208c7b200)
    /usr/src/docker/vendor/src/github.com/godbus/dbus/conn.go:248 +0x58
created by github.com/godbus/dbus.(*Conn).Auth
    /usr/src/docker/vendor/src/github.com/godbus/dbus/auth.go:118 +0xe84

goroutine 20 [chan receive, 1317 minutes]:
github.com/godbus/dbus.(*Conn).outWorker(0xc208c7b200)
    /usr/src/docker/vendor/src/github.com/godbus/dbus/conn.go:370 +0x58
created by github.com/godbus/dbus.(*Conn).Auth
    /usr/src/docker/vendor/src/github.com/godbus/dbus/auth.go:119 +0xea1

goroutine 21 [chan receive, 1317 minutes]:
github.com/docker/libnetwork/iptables.signalHandler()
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/iptables/firewalld.go:92 +0x57
created by github.com/docker/libnetwork/iptables.FirewalldInit
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/iptables/firewalld.go:48 +0x185

goroutine 32 [select]:
github.com/docker/libnetwork/osl.removeUnusedPaths()
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/osl/namespace_linux.go:73 +0x48b
created by github.com/docker/libnetwork/osl.createBasePath
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/osl/namespace_linux.go:58 +0xb1

goroutine 33 [IO wait, 1317 minutes]:
net.(*pollDesc).Wait(0xc2087ef8e0, 0x72, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2087ef8e0, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc2087ef880, 0x0, 0x7f8f366c2500, 0xc208ce8ef0)
    /usr/local/go/src/net/fd_unix.go:419 +0x40b
net.(*UnixListener).AcceptUnix(0xc208cdd900, 0x10, 0x0, 0x0)
    /usr/local/go/src/net/unixsock_posix.go:282 +0x56
net.(*UnixListener).Accept(0xc208cdd900, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/unixsock_posix.go:293 +0x4c
github.com/docker/libnetwork.(*controller).acceptClientConnections(0xc20808c840, 0xc2087ef810, 0x63, 0x7f8f31f8a610, 0xc208cdd900)
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/sandbox_externalkey.go:138 +0x87
created by github.com/docker/libnetwork.(*controller).startExternalKeyListener
    /usr/src/docker/vendor/src/github.com/docker/libnetwork/sandbox_externalkey.go:132 +0x295

goroutine 46 [chan receive, 1317 minutes]:
database/sql.(*DB).connectionOpener(0xc208cfd540)
    /usr/local/go/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
    /usr/local/go/src/database/sql/sql.go:452 +0x31c

goroutine 49 [chan receive]:
github.com/docker/docker/daemon.(*statsCollector).run(0xc208d9adb0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/stats_collector_unix.go:93 +0xb2
created by github.com/docker/docker/daemon.newStatsCollector
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/stats_collector_unix.go:33 +0x116

goroutine 50 [chan receive, 2 minutes]:
github.com/docker/docker/daemon.(*Daemon).execCommandGC(0xc2088a2140)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/exec.go:300 +0x8c
created by github.com/docker/docker/daemon.NewDaemon
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/daemon.go:827 +0x2dbc

goroutine 521 [semacquire, 6 minutes]:
sync.(*Mutex).Lock(0xc2087c0ee0)
    /usr/local/go/src/sync/mutex.go:66 +0xd3
github.com/docker/docker/daemon.(*Daemon).reducePsContainer(0xc2088a2140, 0xc208dc0400, 0xc209c1c3c0, 0xc208c8b4b0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:114 +0x52
github.com/docker/docker/daemon.(*Daemon).reduceContainers(0xc2088a2140, 0xc20a1b4be0, 0xc208c8b4b0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:97 +0x1bc
github.com/docker/docker/daemon.(*Daemon).Containers(0xc2088a2140, 0xc20a1b4be0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:84 +0xa5
github.com/docker/docker/api/server/router/local.(*router).getContainersJSON(0xc208d01540, 0x7f8f31fad118, 0xc20a1e0390, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/container.go:47 +0x3ba
github.com/docker/docker/api/server/router/local.*router.(github.com/docker/docker/api/server/router/local.getContainersJSON)·fm(0x7f8f31fad118, 0xc20a1e0390, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/local.go:109 +0x7b
github.com/docker/docker/api/server.func·004(0x7f8f31fad118, 0xc20a1e0390, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:87 +0x7c7
github.com/docker/docker/api/server.func·003(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:66 +0x10e
github.com/docker/docker/api/server.func·002(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:49 +0x47c
github.com/docker/docker/api/server.func·001(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000, 0xc20a1e01e0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:27 +0x1dc
github.com/docker/docker/api/server.func·007(0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:156 +0x265
net/http.HandlerFunc.ServeHTTP(0xc208cb1740, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000)
    /usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc208c7db30, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000)
    /usr/src/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc20800a780, 0x7f8f31fad0e0, 0xc208da0140, 0xc20b19c000)
    /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208da0000)
    /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
    /usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 70 [chan receive, 1315 minutes]:
github.com/docker/docker/pkg/signal.func·002()
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/signal/trap.go:29 +0x8f
created by github.com/docker/docker/pkg/signal.Trap
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/signal/trap.go:55 +0x250

goroutine 522 [semacquire]:
sync.(*Mutex).Lock(0xc2087c0ee0)
    /usr/local/go/src/sync/mutex.go:66 +0xd3
github.com/docker/docker/daemon.(*Daemon).reducePsContainer(0xc2088a2140, 0xc208dc0400, 0xc209c1c4e0, 0xc20a1954b0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:114 +0x52
github.com/docker/docker/daemon.(*Daemon).reduceContainers(0xc2088a2140, 0xc20a1b4d20, 0xc20a1954b0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:97 +0x1bc
github.com/docker/docker/daemon.(*Daemon).Containers(0xc2088a2140, 0xc20a1b4d20, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/list.go:84 +0xa5
github.com/docker/docker/api/server/router/local.(*router).getContainersJSON(0xc208d01540, 0x7f8f31fad118, 0xc20a1e05a0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/container.go:47 +0x3ba
github.com/docker/docker/api/server/router/local.*router.(github.com/docker/docker/api/server/router/local.getContainersJSON)·fm(0x7f8f31fad118, 0xc20a1e05a0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/local.go:109 +0x7b
github.com/docker/docker/api/server.func·004(0x7f8f31fad118, 0xc20a1e05a0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:87 +0x7c7
github.com/docker/docker/api/server.func·003(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:66 +0x10e
github.com/docker/docker/api/server.func·002(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:49 +0x47c
github.com/docker/docker/api/server.func·001(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0, 0xc20a1e02d0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:27 +0x1dc
github.com/docker/docker/api/server.func·007(0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:156 +0x265
net/http.HandlerFunc.ServeHTTP(0xc208cb1740, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0)
    /usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc208c7db30, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0)
    /usr/src/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc20800a780, 0x7f8f31fad0e0, 0xc208da0b40, 0xc20b19c0d0)
    /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208da0aa0)
    /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
    /usr/local/go/src/net/http/server.go:1751 +0x35e

goroutine 60 [semacquire]:
sync.(*Mutex).Lock(0xc208d39608)
    /usr/local/go/src/sync/mutex.go:66 +0xd3
log/syslog.(*Writer).Close(0xc208d395c0, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:184 +0x4a
github.com/docker/docker/daemon/logger/syslog.(*syslogger).Close(0xc208cc0c58, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/syslog/syslog.go:103 +0x3d
github.com/docker/docker/daemon.(*containerMonitor).resetContainer(0xc20889b110, 0x1376301)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/monitor.go:329 +0x9e4
github.com/docker/docker/daemon.(*containerMonitor).Start(0xc20889b110, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/monitor.go:181 +0x755
github.com/docker/docker/daemon.*containerMonitor.Start·fm(0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container.go:784 +0x39
github.com/docker/docker/pkg/promise.func·001()
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/promise/promise.go:8 +0x2f
created by github.com/docker/docker/pkg/promise.Go
    /usr/src/docker/.gopath/src/github.com/docker/docker/pkg/promise/promise.go:9 +0xfb

goroutine 64 [semacquire]:
sync.(*Mutex).Lock(0xc208d39608)
    /usr/local/go/src/sync/mutex.go:66 +0xd3
log/syslog.(*Writer).writeAndRetry(0xc208d395c0, 0x6, 0xc20850ae40, 0x13, 0x0, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:254 +0x7a
log/syslog.(*Writer).Info(0xc208d395c0, 0xc20850ae40, 0x13, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:240 +0x57
github.com/docker/docker/daemon/logger/syslog.(*syslogger).Log(0xc208cc0c58, 0xc20850c640, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/syslog/syslog.go:99 +0x12f
github.com/docker/docker/daemon/logger.(*Copier).copySrc(0xc208d5a380, 0x1388650, 0x6, 0x7f8f31fac2b8, 0xc208bd3cb0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/copier.go:53 +0x450
created by github.com/docker/docker/daemon/logger.(*Copier).Run
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/copier.go:38 +0x11c

goroutine 65 [IO wait]:
net.(*pollDesc).Wait(0xc20889b250, 0x77, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitWrite(0xc20889b250, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:93 +0x43
net.(*netFD).Write(0xc20889b1f0, 0xc208d92000, 0x292, 0x300, 0x0, 0x7f8f366c2500, 0xc20a1ea1b0)
    /usr/local/go/src/net/fd_unix.go:335 +0x5ee
net.(*conn).Write(0xc208cc0c50, 0xc208d92000, 0x292, 0x300, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/net.go:129 +0xdc
fmt.Fprintf(0x7f8f31facca0, 0xc208cc0c50, 0x1403b70, 0x13, 0xc20a191ba8, 0x6, 0x6, 0x17b, 0x0, 0x0)
    /usr/local/go/src/fmt/print.go:189 +0xac
log/syslog.(*netConn).writeString(0xc208d91d40, 0x1b, 0xc208d9c578, 0x4, 0xc208d91c80, 0x13, 0xc208b06c80, 0x261, 0x1370450, 0x1, ...)
    /usr/local/go/src/log/syslog/syslog.go:295 +0x40a
log/syslog.(*Writer).write(0xc208d395c0, 0x1b, 0xc208b06c80, 0x261, 0x4b7c31, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:277 +0x1bc
log/syslog.(*Writer).writeAndRetry(0xc208d395c0, 0x3, 0xc208b06c80, 0x261, 0x0, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:258 +0xe9
log/syslog.(*Writer).Err(0xc208d395c0, 0xc208b06c80, 0x261, 0x0, 0x0)
    /usr/local/go/src/log/syslog/syslog.go:219 +0x57
github.com/docker/docker/daemon/logger/syslog.(*syslogger).Log(0xc208cc0c58, 0xc20a1b40f0, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/syslog/syslog.go:97 +0xce
github.com/docker/docker/daemon/logger.(*Copier).copySrc(0xc208d5a380, 0x1388610, 0x6, 0x7f8f31fac2b8, 0xc208bd3d40)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/copier.go:53 +0x450
created by github.com/docker/docker/daemon/logger.(*Copier).Run
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/copier.go:38 +0x11c

goroutine 520 [semacquire, 8 minutes]:
sync.(*WaitGroup).Wait(0xc208d5a3a8)
    /usr/local/go/src/sync/waitgroup.go:132 +0x169
github.com/docker/docker/daemon/logger.(*Copier).Wait(0xc208d5a380)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/logger/copier.go:70 +0x34
github.com/docker/docker/daemon.func·037()
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/monitor.go:320 +0x3a
created by github.com/docker/docker/daemon.(*containerMonitor).resetContainer
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/monitor.go:322 +0x8f2

goroutine 519 [semacquire, 8 minutes]:
sync.(*Mutex).Lock(0xc2087c0ee0)
    /usr/local/go/src/sync/mutex.go:66 +0xd3
github.com/docker/docker/daemon.(*State).IsRunning(0xc2087c0ee0, 0xc2089f2bd0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/state.go:160 +0x2f
github.com/docker/docker/daemon.(*Container).Kill(0xc208dc0400, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container.go:463 +0x4f
github.com/docker/docker/daemon.(*Container).Stop(0xc208dc0400, 0xa, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container.go:519 +0x2dd
github.com/docker/docker/daemon.(*Daemon).ContainerStop(0xc2088a2140, 0xc20b079497, 0xc, 0xa, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/stop.go:21 +0x129
github.com/docker/docker/api/server/router/local.(*router).postContainersStop(0xc208d01540, 0x7f8f31fad118, 0xc20a1e0ed0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/container.go:209 +0x1e1
github.com/docker/docker/api/server/router/local.*router.(github.com/docker/docker/api/server/router/local.postContainersStop)·fm(0x7f8f31fad118, 0xc20a1e0ed0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/local.go:135 +0x7b
github.com/docker/docker/api/server.func·004(0x7f8f31fad118, 0xc20a1e0ed0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:87 +0x7c7
github.com/docker/docker/api/server.func·003(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:66 +0x10e
github.com/docker/docker/api/server.func·002(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:49 +0x47c
github.com/docker/docker/api/server.func·001(0x7f8f366c2a50, 0xc20802afd0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410, 0xc20a1e0c60, 0x0, 0x0)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:27 +0x1dc
github.com/docker/docker/api/server.func·007(0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410)
    /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:156 +0x265
net/http.HandlerFunc.ServeHTTP(0xc208926ec0, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410)
    /usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc208c7db30, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410)
    /usr/src/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc20800a780, 0x7f8f31fad0e0, 0xc208da0dc0, 0xc20b19c410)
    /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208da0d20)
    /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
    /usr/local/go/src/net/http/server.go:1751 +0x35e
```

Also all docker processes are waiting for something according to strace:

```
strace -p 18143
Process 18143 attached
epoll_wait(5, 
^CProcess 18143 detached
 <detached ...>
strace -p 11935
Process 11935 attached
futex(0x1cd0678, FUTEX_WAIT, 0, NULL
^CProcess 11935 detached
 <detached ...>
strace -p 12011
Process 12011 attached
futex(0x1cd0678, FUTEX_WAIT, 0, NULL^CProcess 12011 detached
 <detached ...>
```

**Comment by HackToday** [comment]
Did you container special or any other container can reproduce such issue ?  Thanks

**Comment by vpaul** [comment]
Nope, not a like special. It is reproducible on different instances with different containers.  aufs/devmapper doesn't matter.

**Comment by thaJeztah** [comment]
@vpaul are you still having this issue? Can you try upgrading to a more recent version toes if it's fixed? We no longer do patch releases for 1.9, but interested to hear if this is still an issue on the current release.  I'll close this issue for now, but feel free to comment after

**Comment by iaon** [comment]
I think it was related to https://github.com/docker/docker/issues/18180
After kernel upgrade we didn't met such behaviour.
